kernal/drivers/x16/rtc.s

```diff
@@ -156,3 +162,67 @@ bin_to_bcd:
 @end:	cld
 	ply
 	rts
+
+; Inputs: 
+; Y = nvram offset
+; A = byte value (for write)
+;
+; Outputs:
+; A = byte value (for read)
+; C = 0: success
+; C = 1: failure
+;
+; clobbers X
+rtc_get_nvram:
+	clc
+	bra rtc_nvram
+rtc_set_nvram:
+	sec
+rtc_nvram:
+	php
+	pha
+
+	tya
+	and #$3f
+	clc
+	adc #nvram_base
+	tay
+	pla
+	ldx #rtc_address
+	plp
+	bcs @write
+	jmp i2c_read_byte
+@write:
+    jsr i2c_write_byte
+	bcs @exit
+	cpy #screen_mode_cksum_addr
+	bcc @checksum
+@good:
+	clc
+@exit:
+	rts
+@checksum:
+	jsr rtc_check_nvram_checksum
+	bcs @exit
+	lda tmp2
+	jmp i2c_write_byte ; commit the new checksum
+
+; sets Z if equal, C on i2c error
+rtc_check_nvram_checksum:
+	ldx #rtc_address
+	ldy #nvram_base
+	stz tmp2
+@cksumloop:
+	jsr i2c_read_byte
```

Comment:
If you want to speed up this it's possible to use the functions i2c_read_first_byte/i2c_read_next_byte and i2c_read_stop instead.